.circleci/config.yml

```diff
@@ -1,8 +1,184 @@
 version: 2.1
 
+orbs:
+  slack: circleci/slack@3.3.0
+
+defaults:
+  workspace_root: &workspace_root
+                    ~/falcon
+  # Default machine settings.
+  default_machine: &default_machine
+    working_directory: *workspace_root
+    machine:
+      enabled: true
+      image: ubuntu-1604:201903-01
+
+commands:
+  hosting_cli:
+    description: "Execute Hosting CLI"
+    parameters:
+      command:
+        type: string
+      title:
+        type: string
+    steps:
+      - run:
+          name: << parameters.title >>
+          command: |
+            docker login -u $DOCKER_HUB_USER -p $DOCKER_HUB_PASSWORD &&
+            docker run \
+              -v ${PWD}/.deployment:/.deployment:ro \
+              -v ${PWD}/.env:/.project-env-file \
+              -e AWS_SECRET_ACCESS_KEY="$AWS_SECRET_ACCESS_KEY" \
+              -e AWS_ACCESS_KEY_ID="$AWS_ACCESS_KEY_ID" \
+              -e AWS_DEFAULT_REGION="$AWS_DEFAULT_REGION" \
+              -e AWS_DEFAULT_PROFILE="$AWS_DEFAULT_PROFILE" \
+              -e AWS_DEFAULT_ROLE="$AWS_DEFAULT_ROLE" \
+              systemseed/hosting-cli:latest << parameters.command >>
+
+  drupal_post_deploy:
+    description: "Perform Drupal post deployment steps"
+    parameters:
+      chart:
+        type: string
+      component:
+        type: string
+      title:
+        type: string
+    steps:
+      - hosting_cli:
+          title: << parameters.title >>
+          command: /.deployment/scripts/post-deploy.sh << parameters.chart >> << parameters.component >> $CIRCLE_BRANCH $TEST_USER_PASSWORD
+
+  urls_to_env:
+    description: "Prints URLs to project .env file"
+    steps:
+      - hosting_cli:
+          title: "Prints URLs to project .env file"
+          command: /.deployment/scripts/urls_to_file.sh $CIRCLE_BRANCH /.project-env-file
+
+
 jobs:
-  build_and_test:
+  ######################################################
+  # PREPARE CIRCLE CI WORKSPACE TO REUSE IN OTHER JOBS #
+  ######################################################
+  prepare_workspace_for_aws:
+    <<: *default_machine
+
+    steps:
+      - checkout
+
+      - run:
+          name: Fix CircleCI files permissions
+          command: |
+            sudo chown ubuntu:ubuntu -R *
+            sudo chown ubuntu:ubuntu -R .[!.]*
+
+      - run:
+          name: Generate .env and docker-compose files
+          command: |
+            make down
+
+      - run:
+          name: Install node dependencies (without docker-compose up)
+          command: |
+            docker-compose run --no-deps --rm node yarn install
+
+      - run:
+          name: Install php dependencies (without docker-compose up)
+          command: |
+            docker-compose run --no-deps --user 0:0 --rm php bash -c "composer install --no-suggest"
+            docker-compose run --no-deps --user 0:0 --rm php bash -c "chmod +w web/sites/default"
```

Comment:
Why do we need to run it from root user? Default wodby user seems to be more suitable for it.

Reply:
Removed --user 0:0.